default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,201 @@
 # Summary
 [summary]: #summary
 
-One para explanation of the feature.
+Allow struct definitions to supply default values for individual fields,
+and then allow those fields to be omitted from struct initialisation:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+}
+```
 
 # Motivation
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Today, Rust allows you to create an instance of a struct using a literal syntax.
+This requires all fields in the struct be assigned a value, and can't be used
+with private fields.
+It can also be inconvenient for large structs that usually receive the same values.
+
+With the `..` syntax, values for missing fields can be taken from another struct.
+This is convenient for using `Default` to override just a few values on initialisation:
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+    ..Default::default()
+}
+```
+
+However, this still requires an already initialised struct after the `..`.
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that you need one for each combination of fields a caller can supply.
```

Comment:
Or supply one constructor with mandatory values and expect the user to override public default values after construction.
```rust
let foo = Foo::new(42); // sets `foo.c`
foo.b = true; // override b
```